### PR TITLE
Add the new validation step interfaces

### PR DESCRIPTION
**Pull request by loic-sharma**
This adds the new validation step interfaces that will process "generic" validations. These interfaces will be used by the Marketplace team to implement the new content scanning validation step.

These new interfaces aren't "connected" to the Orchestrator yet. This will be added in the future (tracked by https://github.com/NuGet/Engineering/issues/3583)

Spec: https://github.com/NuGet/Engineering/blob/master/Server.Specs/OrchestratorUnifiedValidations.md#new-orchestrator-interfaces

Part of: https://github.com/NuGet/Engineering/issues/3583

/cc @anupkmr03 @nikhgup @rheajain

### CODE REVIEW

File: src/Validation.Common.Job/Validation/IProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Services.Validation
+{
+    /// <summary>
+    /// A marker interface to state that this validator mutates packages. The only purpose of this interface it to
+    /// allow the caller of the validators (e.g. orchestrator) to verify in advance that a validator (i.e. processor)
+    /// that mutates packages does not run in parallel with any other validator.
+    /// </summary>
+    public interface IProcessor : IValidator
```
Comment (loic-sharma) [inline src/Validation.Common.Job/Validation/IProcessor.cs:11]: Mirrors:

https://github.com/NuGet/NuGet.Jobs/blob/35e9eecf0cad3d97cbbbc51076430b3a11e4336d/src/Validation.Common.Job/Validation/INuGetProcessor.cs#L1-L14

File: src/Validation.Common.Job/Validation/IProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    {
+    }
+}
```
Comment (loic-sharma) [inline src/Validation.Common.Job/Validation/IProcessor.cs:14]: Mirrors:

https://github.com/NuGet/NuGet.Jobs/blob/35e9eecf0cad3d97cbbbc51076430b3a11e4336d/src/Validation.Common.Job/Validation/INuGetProcessor.cs#L1-L14

File: src/Validation.Common.Job/Validation/IValidationRequest.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        /// <summary>
+        /// Deserializes the validation set's properties content as JSON.
+        /// </summary>
+        /// <seealso cref="PackageValidationSet.ValidationProperties"/>
+        /// <typeparam name="T">The type to use to deserialize the properties.</typeparam>
+        /// <returns>The validation set's properties.</returns>
+        T GetProperties<T>();
+    }
+}
```
Comment (loic-sharma) [inline src/Validation.Common.Job/Validation/IValidationRequest.cs:33]: Mirrors:

https://github.com/NuGet/NuGet.Jobs/blob/35e9eecf0cad3d97cbbbc51076430b3a11e4336d/src/Validation.Common.Job/Validation/INuGetValidationRequest.cs#L1-L42

File: src/Validation.Common.Job/Validation/IValidationResponse.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        /// </summary>
+        string OutputUrl { get; }
+
+        /// <summary>
+        /// The results generated by the validation step. These provide additional information,
+        /// such as debugging information or reasons for failure.
+        /// </summary>
+        IReadOnlyList<PackageValidationResult> Results { get; }
+    }
+}
```
Comment (loic-sharma) [inline src/Validation.Common.Job/Validation/IValidationResponse.cs:31]: Mirrors:

https://github.com/NuGet/NuGet.Jobs/blob/35e9eecf0cad3d97cbbbc51076430b3a11e4336d/src/Validation.Common.Job/Validation/INuGetValidationResponse.cs#L1-L30

File: src/Validation.Common.Job/Validation/IValidator.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        /// <summary>
+        /// A method that can be used to clean up any state produced by <see cref="StartAsync(IValidationRequest)"/>.
+        /// If this method fails for some reason, an exception should be thrown. However, the caller can simply swallow
+        /// the exception making this method a best effort. This is simply a way, in the happy path, to keep storage
+        /// usage under control.
+        /// </summary>
+        /// <param name="request">The validation step request.</param>
+        Task CleanUpAsync(IValidationRequest request);
+    }
+}
```
Comment (loic-sharma) [inline src/Validation.Common.Job/Validation/IValidator.cs:54]: Mirrors:

https://github.com/NuGet/NuGet.Jobs/blob/35e9eecf0cad3d97cbbbc51076430b3a11e4336d/src/Validation.Common.Job/Validation/INuGetValidator.cs#L1-L54

File: src/Validation.Common.Job/Validation/IProcessor.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Services.Validation
+{
+    /// <summary>
+    /// A marker interface to state that this validator mutates packages. The only purpose of this interface it to
```
Comment (zhhyu) [inline src/Validation.Common.Job/Validation/IProcessor.cs:7]: nit: "The only purpose of this interface is to"?

Comment (loic-sharma) [reply src/Validation.Common.Job/Validation/IProcessor.cs]: Good catch! Fixed :)

**Review by joelverhagen — APPROVED** [review]

**Pull request merged**